src/tests/ftest/util/dfuse_utils.py

```diff
@@ -121,15 +122,16 @@ def check_mount_state(self, nodes=None):
             for host in hosts:
                 if retcode == 0:
                     check_mounted.add(host)
-                else:
-                    command = "grep 'dfuse {}' /proc/mounts" .format(self.mount_dir.value)
-                    retcodes = pcmd([host], command, expect_rc=None)
-                    for ret_code, host_names in list(retcodes.items()):
-                        for node in host_names:
-                            if ret_code == 0:
-                                check_mounted.add(node)
-                            else:
-                                state["nodirectory"].add(node)
+                    continue
+
+                command = "grep 'dfuse {}' /proc/mounts" .format(self.mount_dir.value)
+                retcodes = pcmd([host], command, expect_rc=None)
+                for ret_code, host_names in list(retcodes.items()):
+                    for node in host_names:
+                        if ret_code == 0:
+                            check_mounted.add(node)
+                        else:
+                            state["nodirectory"].add(node)
```

Comment:
FYI, I have a PR with a pylint fix for this as well: https://github.com/daos-stack/daos/pull/7822/files

Reply:
I wasn't sure yours was correct, but I'll remove this part from my PR.

Reply:
I did verify mine by running most dfuse tests, and by checking the logic separately. The big win here is that `NodeSet.add(Nodeset())` works. So the inner loop can become a single call.